### PR TITLE
Add new schedule primitive

### PR DESCRIPTION
**Pull request by haozech**
此pr中包含了调度原语重构中，流程切换的工作。目前通过flag控制是否使用新的schedule，且新的schedule只适用于使用新的op fusion pass的情况下。开发还没有全部完成，目前只有部分op完成了切换。

### CODE REVIEW

File: cinn/common/cas.cc [hunk 1]
```diff
@@ -1325,7 +1325,9 @@ Expr CasSimplifyMutator::SimplifyMod(Expr u) {
       for (int i = 0; i < sum_args.size(); i++) {
         auto* arg_var = sum_args[i].As<_Var_>();
         all_positive_var =
-            all_positive_var && arg_var && var_intervals.count(arg_var->name) && var_intervals.at(arg_var->name).l >= 0;
+            all_positive_var && arg_var &&
```
Comment (zhhsplendid) [inline cinn/common/cas.cc]: In the future, try to put long boolean expression as a function.

File: cinn/hlir/pe/new_schedule.cc [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+#include "cinn/hlir/pe/load_x86_params.h"
+#include "cinn/hlir/pe/schedule.h"
+#include "cinn/optim/ir_simplify.h"
+#include "cinn/poly/isl_utils.h"
+
+namespace cinn {
+namespace hlir {
+namespace pe {
+
+void NewScheduleInjectiveCPU(ir::IRSchedule &ir_sch,
```
Comment (zhhsplendid) [inline cinn/hlir/pe/new_schedule.cc]: Try not to name functions and variables with "new" to represent a new version.

1. Base on my experience, you would change the "new variables/functions" after several months during software development, then you have to call "NewNew", "NewNewNew", and the "new" will be meaning "old" at that time.

2. "new" has the special meaning of creating an new object in C++ software engineering. So some engineers would be confused when they read a void "NewXXX" function, they would think something is created here.

File: cinn/hlir/pe/new_schedule.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 CINN Authors. All Rights Reserved.
```
Comment (zhhsplendid) [inline cinn/hlir/pe/new_schedule.h]: Copyright 2022?

File: cinn/hlir/pe/new_schedule.cc [hunk 4]
```diff
@@ -0,0 +1,543 @@
+// Copyright (c) 2021 CINN Authors. All Rights Reserved.
```
Comment (zhhsplendid) [inline cinn/hlir/pe/new_schedule.cc]: Copyright 2022?

File: cinn/hlir/pe/new_schedule.cc [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+}
+
+void NewCudaSplitSchedule(ir::IRSchedule &ir_sch,
+                          const std::vector<std::vector<int>> &output_shapes,
+                          int axis,
+                          const common::Target &target) {
+  ir_sch.MergeExprs();
+  int dims = output_shapes[0].size();
+  std::vector<int> reorders;
+  for (int i = 0; i < dims; i++) {
```
Comment (zhhsplendid) [inline cinn/hlir/pe/new_schedule.cc]: Prefer `++i` in the future, reference:

https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement

Comment (haozech) [reply cinn/hlir/pe/new_schedule.cc]: Done, thanks

File: cinn/hlir/pe/new_schedule.h (hunk 3 above)
Comment (haozech) [reply cinn/hlir/pe/new_schedule.h]: Done, thanks

File: cinn/hlir/pe/new_schedule.cc (hunk 4 above)
Comment (haozech) [reply cinn/hlir/pe/new_schedule.cc]: Done, thanks

File: cinn/hlir/pe/new_schedule.cc (hunk 2 above)
Comment (haozech) [reply cinn/hlir/pe/new_schedule.cc]: Done, thanks

File: cinn/common/cas.cc (hunk 1 above)
Comment (haozech) [reply cinn/common/cas.cc]: Done, thanks

**Review by wenming2014 — DISMISSED** [review]

**Review by wenming2014 — APPROVED** [review]

**Pull request merged**